Generator/XsdToDtoGenerator.php

```diff
@@ -0,0 +1,202 @@
+    }
+    
+    /**
+     * Gets the source
+     * 
+     * @return string
+     */
+    public function getSource() 
+    {
+        return $this->source;
+    }
+
+    /**
+     * Sets the destination 
+     * 
+     * @param string $destination
+     * 
+     * @return XsdToDtoGenerator
+     */
+    public function setDestination($destination) 
+    {
+        $this->destination = $destination;
+        return $this;
+    }
+    
+    /**
+     * Gets the destination
+     * 
+     * @return string
+     */
+    public function getDestination() 
+    {
+        return $this->destination;
+    }
+    
+    /**
+     * Sets the destination Namespace
+     * 
+     * @param string $destinationNS
+     * 
+     * @return XsdToDtoGenerator
+     */
+    public function setDestinationNS($destinationNS) 
+    {
+        $this->destinationNS = $destinationNS;
+        return $this;
+    }
+
+    /**
+     * Gets the Destination Namespace
+     * 
+     * @return string
+     */
+    public function getDestinationNS() 
+    {
+        return $this->destinationNS;
+    }
+
+    /**
+     * Generates the DTOs
+     * 
+     * @throws InvalidArgumentException
+     */
+    public function generate($forceOverwrite = true, $format = 'php')
+    {
+        $this->setFormat($format);
+        $source = $this->getSource();
+        if (!$source) {
+            throw new InvalidArgumentException('Source');
+        }
+        
+        $destination = $this->getDestination();
+        if (!is_dir($destination)) {
+            throw new InvalidArgumentException('Destination');
+        }
+        
+        $destinationNS = $this->getDestinationNS();
+        if (!$destinationNS) {
+            throw new InvalidArgumentException('DestinationNS');
+        }
+
+        $reader = new XsdReader();
+        $reader->read($source);
+        
+        $firstElement = $reader->getFirstElementWithChildren();
+        $this->genereateDTOClasses($firstElement, $forceOverwrite);
+    }
+    
+    /**
+     * Returns a target file
+     * 
+     * @return string
+     */
+    protected function setTargetFile($elementName)
+    {
+        $this->target = sprintf("%s/%s.%s", $this->destination, $elementName, $this->format);
+        
+        return $this;
+    }
+
+    /**
+     * Sets the configuration format.
+     *
+     * @param string $format The configuration format
+     */
+    protected function setFormat($format)
+    {
+        $this->format = $format;
+    }
+    
+    
+    protected function genereateDTOClasses($firstElementWithChildren, $forceOverwrite)
+    {
+        foreach ($firstElementWithChildren->getChildren() as $element) {
+            if ($element instanceof Reader\Xsd\ComplexTypeElement) {
+                $element->setDataType($this->getTypeForChildDto($element));
+                $this->genereateDTOClasses($element, $forceOverwrite);
+            }
+        }
+        $this->setTargetFile(ucfirst($firstElementWithChildren->getName()));
```

Comment:
Wasn't there a method in that class - ```getUcfirstName``` which does that uppercasing. Personally I would use it as it is here and remove the method in the class, as this is a modifier which generally should not be part of it.